RezoningScraper/CacheManager.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Polly.Caching;
+using System.IO.IsolatedStorage;
+using System.Text.Json;
+using System.Security.Cryptography;
+using System.Text;
+using static Spectre.Console.AnsiConsole;
+
+namespace RezoningScraper
+{
+    public class CacheManager<TResult> : IAsyncCacheProvider<TResult>
+    {
+        record CachedRecord<T>(DateTime Ttl, T Record);
+        public async Task PutAsync(string key, TResult value, Ttl ttl, CancellationToken ct, bool continueOnCapturedContext)
+        {
+            using IsolatedStorageFile isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Domain | IsolatedStorageScope.Assembly, null, null);
```

Comment:
This is interesting, I haven't used [isolated storage](https://docs.microsoft.com/en-us/dotnet/standard/io/isolated-storage) before. This bit from the docs gives me pause:

> This is a legacy component primarily designed for now-deprecated Code Access Security scenarios.

The docs are also unclear on how/whether it works on non-Windows platforms.

I'm OK using it if it does the trick, but I'd like to spend a little bit of time understanding it better first. An alternative might be to dump it in the database.

Reply:
Isolated storage is not an API with a future, so if you are comfortable putting the cache in SQLite, I'll move it there.

Reply:
Sounds good.

I don't have a strong opinion on what the schema should look like; could be in its own table, or we could create a shared cache table for tokens and API responses.

Reply:
The cache table is a generic key-value cache with customizable expiry so you could move the token in there if you wanted.